### PR TITLE
Fix  CRD import issue with `oneOf`, `not`, and `anyOf`

### PR DESCRIPTION
**Pull request by errordeveloper**
CUE now has resolved the CRD import issue with `oneOf`, `not`, and `anyOf` (#336) as well required fields. This PR integrates latest revision and updates tests.

Fix: #336

### CODE REVIEW

File: cmd/timoni/testdata/crd/golden/cue.mod/gen/flagger.app/alertprovider/v1beta1/types_gen.cue [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+		// Name of the Kubernetes secret
+		name!: string
+	}
+
+	// Type of this provider
+	type?: "slack" | "msteams" | "discord" | "rocket" | "gchat"
+
+	// Bot username for this provider
+	username?: string
+}
```
Comment (errordeveloper) [inline cmd/timoni/testdata/crd/golden/cue.mod/gen/flagger.app/alertprovider/v1beta1/types_gen.cue:69]: @stefanprodan I suppose this one might be small enough to eyeball? 😅

Comment (stefanprodan) [reply cmd/timoni/testdata/crd/golden/cue.mod/gen/flagger.app/alertprovider/v1beta1/types_gen.cue:69]: Yeap, looks great!

**Comment by errordeveloper** [comment]
Adding a bump for gomega, as it gets only a shy +1 as part of CUE bump, might as well get the very latest.

File: go.mod [hunk 2]
```diff
@@ -3,7 +3,7 @@ module github.com/stefanprodan/timoni
 go 1.22.0
 
 require (
-	cuelang.org/go v0.9.2
+	cuelang.org/go v0.11.0-alpha.1.0.20240910060539-63413629e06b
```
Comment (stefanprodan) [inline go.mod:6]: We'll need to wait for v0.11.0 to go GA before merging.

**Comment by errordeveloper** [comment]
Should we add test cases for all verbs? I think Flagger CRD has only one of them.

File: go.mod (hunk 2 above)
Comment (errordeveloper) [reply go.mod:6]: Yes... only if one could use build tags in 'go.mod'!

**Comment by stefanprodan** [comment]
v0.11.0 is out so we can move forward with this PR https://github.com/cue-lang/cue/releases/tag/v0.11.0

**Comment by errordeveloper** [comment]
Just did a quick bump, let's see if CI is happy, will rebase then.

**Comment by stefanprodan** [comment]
@errordeveloper I don't think CI would run if the PR has conflicts, but you can run it locally with `make test`.

**Comment by errordeveloper** [comment]
@stefanprodan tests are passing now now, please have a look over the changes, I cannot be too sure of what some of the CUE changes in tests amount to, clearly a few things have changes in 0.11.0.

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @errordeveloper 🏅

**Pull request merged**